sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeLocalShuffleReader.scala

```diff
@@ -28,17 +28,19 @@ import org.apache.spark.sql.execution.joins.{BroadcastHashJoinExec, BuildLeft, B
 import org.apache.spark.sql.internal.SQLConf
 
 object BroadcastJoinWithShuffleLeft {
-  def unapply(plan: SparkPlan): Option[(QueryStageExec, BuildSide)] = plan match {
-    case join: BroadcastHashJoinExec if ShuffleQueryStageExec.isShuffleQueryStageExec(join.left) =>
-      Some((join.left.asInstanceOf[QueryStageExec], join.buildSide))
+  def unapply(plan: SparkPlan): Option[(SparkPlan, BuildSide)] = plan match {
+    case join: BroadcastHashJoinExec if join.left.isInstanceOf[QueryStageExec]
```

Comment:
shall we use `ShuffleQueryStageExec.isShuffleQueryStageExec(join.left)`? `join.left.isInstanceOf[QueryStageExec]` can be broadcast exchange.